cmake/StaticBuild.cmake

```diff
@@ -23,7 +23,31 @@ if(BUILD_STATIC)
 
   # gettext is included in libc on many unix systems
   if(NOT LIBC_HAS_DGETTEXT)
-    set(GETTEXT_LIBRARIES "-Wl,-Bstatic -lintl -liconv -Wl,-Bdynamic")
+    FIND_LIBRARY(IDN2_LIBRARY NAMES idn2 libidn2
+      HINTS ${PC_GETTEXT_LIBDIR} ${PC_GETTEXT_LIBRARY_DIRS})
+    FIND_LIBRARY(UNISTRING_LIBRARY NAMES unistring libunistring
+      HINTS ${PC_GETTEXT_LIBDIR} ${PC_GETTEXT_LIBRARY_DIRS})
+
+    set(GETTEXT_LIBRARIES "-Wl,-Bstatic")
+
+    if(IDN2_LIBRARY)
+      set(GETTEXT_LIBRARIES "${GETTEXT_LIBRARIES} -lidn2")
+    endif()
+
+    set(GETTEXT_LIBRARIES "${GETTEXT_LIBRARIES} -lintl")
+
+    if(UNISTRING_LIBRARY)
+      set(GETTEXT_LIBRARIES "${GETTEXT_LIBRARIES} -lunistring")
+    endif()
+    
+    set(GETTEXT_LIBRARIES "${GETTEXT_LIBRARIES} -liconv")
+
+    set(GETTEXT_LIBRARIES "${GETTEXT_LIBRARIES} -Wl,-Bdynamic")
+
+    if(CMAKE_GENERATOR STREQUAL "MinGW Makefiles" AND UNISTRING_LIBRARY)
```

Comment:
We don't need to check what generator is used. This issue should appear for any build system.

Reply:
I check the generator because the MSYS2 MinGW-w64 static libraries introduce a dynamic link dependency on the unistring library.  I don't want to break the official build process.  Do you know a better way to check it?  Should I just dynamically link for all builds?  I can't check the MXE build process right now.

Reply:
I changed it to dynamic linking for all builds.